### ISSUE TITLE
Make Ctrl-W configurable or escapable

### ISSUE THREAD

**Issue by chrisbouchard** [austinjones/tab-rs#183]
The Ctrl-W sequence is used by some terminal commands&nbsp;&mdash; in my case, by Vim for window management. It would be good to have a way to do at least one of the following:

1. Change the Ctrl-W binding to a different sequence,
2. Bind a different sequence to send a literal Ctrl-W (similar to Ctrl-A A in screen), _or_
3. Disable the Ctrl-W binding (possibly conditionally).

My personal preference would be for 1 or 2 (or both). I tried to look around in the code to see if this was already possible, but I was unable to find the code that recognizes a Ctrl-W sequence.

**Comment by austinjones** [comment]
@chrisbouchard Interesting.  I'll need to think about how to handle this.  I had thought that `termwinkey` could be used to rebind vim, but it doesn't look like it works for the split/switch commands.

The code that recognizes the Ctrl-W is here: https://github.com/austinjones/tab-rs/blob/main/tab-command/src/service/terminal/echo_mode.rs#L81

I may add a global config that can be used to rebind the escape sequence.

**Comment by solarizedalias** [comment]
Hi,
I want to +1 for disabling the bind with user configs because there are too many things which expect a certain key-sequence and it's difficult to avoid clashes for all those. It should be optional to use key-bindings or not.
Personally, it feels inconsistent that you can do any other operation (starting, closing, listing) with commands `tab <something>` but you have to use a keybind to disconnect.
(And I was wondering with #188, you could do things like `alias bye='if [ -n "$TAB" ];then tab --disconnect; else exit; fi'`)

**Comment by austinjones** [comment]
Yeah, I agree!  I'm definitely adding this and #188 to the next release.

And yes, it should work in aliases and scripts.  If you don't care about the exit code then `alias bye='tab --disconnect'` would work too.

**Comment by solarizedalias** [comment]
My point was that you wouldn't have to care about whether you're *in tab* or not, you just always say `bye` to the terminal when leaving (btw in zsh, `bye` is actually a builtin which does the same as `exit`).
But I see some cases where key-bindings are more useful (already running another command, hangups, etc) so the both has its own place.
I'm looking forward to the next release anyway, thanks!

**Comment by austinjones** [comment]
@solarizedalias thanks for all the info, it was really helpful!  

On main I've implemented `tab --disconnect` / `tab -z`, which can be run as a command, or within an alias/script.  When invoked with no args, it will disconnect the active session.  The shell alias `bye = tab --disconnect || exit` should do what you want.  `tab --close` / `tab -w` now also close the current tab if no args are provided.

After looking at emacs/vim/other editor bindings, it seems like `ctrl-W` was a poor default.  I'm working on changing it to `ctrl-T`, which triggers the fuzzy finder I added in #1.  I'm also adding a global config file that can rebind the disconnect/fuzzy sequences.

I think I'll have the release ready this weekend if all the testing goes well.
